bot/main.py

```diff
@@ -46,44 +47,119 @@ async def deletevc(ctx, vc_name):
     for channel in category.channels:
         if len(channel.members) == 0:
             await channel.delete()
+            
+    if len(category.channels) == 0:
+        await category.delete
 
+@client.event
+async def on_voice_state_update(member, before, after):
+    if (not before.channel and after.channel):
+        category = discord.utils.get(member.guild.categories, name="Resume Waiting")
+        if (category and discord.utils.get(category.channels, name=member.voice.channel.name)):
+            voice_channel = member.voice.channel.name
+            channel = discord.utils.get(member.guild.channels, name = "resume-bot", type=discord.ChannelType.text)
+            if voice_channel.lower() in client.list_of_queues:
+                if client.lock_queue_from_list[voice_channel.lower()] == True:
+                    await channel.send(
+                        "Queue has been locked, could not be added"
+                    )
+                    await member.move_to(None)
+                else: 
+                    client.list_of_queues[voice_channel.lower()].append(member)
+            else:
+                new_queue = [member]
+                client.list_of_queues[voice_channel.lower()] = new_queue
+                client.lock_queue_from_list[voice_channel.lower()] = False
+            
+            if client.lock_queue_from_list[voice_channel.lower()] == False:
+                await channel.send(
+                    f"Added {member.mention} to queue: {voice_channel}"
+                )
+
+@client.command(name="clear", help="clean up everything when bot is finished being used")
+async def clear(ctx):
+    await deletevc(ctx, "Temporary VC")
+
+# @client.command(name="join", help="get added to queue")
```

Comment:
delete comment if not being used. you can always go back in the git history to find the code